app/controllers/application_controller.rb

```diff
@@ -1008,10 +1008,17 @@ def add_noindex_header_to_non_canonical
   end
 
   def set_cross_origin_opener_policy_header
+    referrer_host =
+      begin
+        URI(request.referrer).host
+      rescue StandardError
+        nil
+      end
```

Comment:
If `SiteSetting. cross_origin_opener_unsafe_none_referrers ` is blank because it is not set, we should not have to spend time parsing the host from the referral because the referral_host will not be used.